### PR TITLE
workflows: WIP initial commit

### PR DESCRIPTION
**Pull request by jalavik**
* Initial port of workflows module with associated tasks.

* Adds `audit` module to track user actions in workflows interface.

* Adds `converter` module for XML conversion using XSLT stylesheets.

NOTE: things will be moved around still, here is current TODO:

- [ ] Reenable commented out tasks for literature and hep ingestion workflows
- [ ] Seperate out logically seperate commits (e.g. setup updates)
- [ ] Update and add tests
- [ ] Update integration together with invenio-workflows-ui
- [ ] Create official repo for invenio-workflows-ui and release

### CODE REVIEW

File: inspirehep/modules/workflows/models.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        """Create a new Payload object."""
+        self.files = []
+        self.drafts = {}
+        self.type = self.get_type(type)
+        self.title = ''
+        self.sips = []
+        super(Payload, self).__init__(workflow_object, self.type, user_id)
+
+    @classmethod
+    def get_type(self, type_or_id):
```
Comment (jmartinm) [inline inspirehep/modules/workflows/models.py]: should the first parameter be `cls`?

File: inspirehep/modules/workflows/receivers.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+@oaiharvest_finished.connect
+def spawn_workflow_from_oai_harvest(request, records, name, **kwargs):
+    """Receive a list of harvested OAI-PMH records and schedule workflow."""
+    from flask import current_app
+    from invenio_workflows import start, workflows
+    # OAIHARVESTER_WORKFLOWS = {
+    #     "http://export.arxiv.org/oai2": "oaiarXiv2marcxml.xsl",
+    # }
+    workflow = "hep_ingestion"
+    import ipdb; ipdb.set_trace()
```
Comment (jmartinm) [inline inspirehep/modules/workflows/receivers.py]: whoops

File: inspirehep/modules/workflows/tasks/submission.py [hunk 3]
```diff
@@ -97,3 +289,104 @@ def _close_ticket(obj, eng):
...
+        combined_callback_url = os.path.join(current_app.config["SERVER_NAME"], callback_url)
+        model = eng.workflow_definition.model(obj)
+        sip = model.get_latest_sip()
+        marcxml = sip.package
+        result = make_robotupload_marcxml(
+            url=url,
+            marcxml=marcxml,
+            callback_url=combined_callback_url,
+            mode=mode,
+            nonce=obj.id
```
Comment (kaplun) [inline inspirehep/modules/workflows/tasks/submission.py]: Can you add `priority=5` as in 39ff6d987879a3a8e38077cda281c8c03736e313

Comment (kaplun) [inline inspirehep/modules/workflows/tasks/submission.py]: On `production` branch I had to amend this for `authors` and for `suggestion`. Can you check if this has now changed and this is the only place where  `make_robotupload_marcxml` need to be amended?

Comment (jalavik) [inline inspirehep/modules/workflows/tasks/submission.py]: Yep, I will check.

**Comment by kaplun** [comment]
@jalavik this PR needs to be closed and reopened against `master`.

**Pull request closed without merging**